packages/application-shell/src/components/navbar/use-navbar-state-manager.ts

```diff
@@ -119,6 +119,47 @@ const useNavbarStateManager = (props: HookProps) => {
           })
           .filter(nonNullable)
       : [];
+  const organizationCustomAppsMenu =
+    projectExtensionsQuery &&
+    projectExtensionsQuery.projectExtension &&
+    projectExtensionsQuery.projectExtension.installedApplications
+      ? projectExtensionsQuery.projectExtension.installedApplications
+          .map<TNavbarMenu | undefined>((installedApplication) => {
+            const application = installedApplication.application;
+            // Map the menu properties to match the one from the proxy schema.
+            // This is to ensure that the menu object is the same from the proxy
+            // config and the custom apps config, thus allowing them to be
+            // concatenated and rendered the same way.
+            if (!application.menuLinks) return;
+            return {
+              key: application.id,
+              uriPath: application.entryPointUriPath,
+              labelAllLocales: application.menuLinks.labelAllLocales || [],
+              icon: application.menuLinks.iconName,
+              permissions: application.menuLinks.permissions as string[],
+              defaultLabel: application.menuLinks.defaultLabel,
```

Comment:
FYI: When rendering the menu link component, we should also use this value in case there is no matching locale.

Reply:
incoming ;)

Reply:
allApplicationNavbarMenu